autotest/src/main/kotlin/no/nav/familie/tilbake/e2e/domene/builder/KravgrunnlagBuilder.kt

```diff
@@ -0,0 +1,171 @@
+                          muligforeldelse: Boolean,
+                          periodeLengde: Int) {
+
+    // Kommer til å trenge å kunne sette kontrollfelt tilbake i tid for at den plukkes av auto-opprett batch (ikke laget enda)
+    private val finalKontrollfelt = LocalDateTime.now().format(DateTimeFormatter.ofPattern("yyyy-MM-dd-HH.mm.ss.SSSSSS"))
+
+    private val request = DetaljertKravgrunnlagMelding().also { detaljertKravgrunnlagMelding ->
+        detaljertKravgrunnlagMelding.detaljertKravgrunnlag = DetaljertKravgrunnlagDto().apply {
+            this.kravgrunnlagId = kravgrunnlagId ?: Random.nextInt(100000, 999999).toBigInteger()
+            this.vedtakId = vedtakId ?: Random.nextInt(100000, 999999).toBigInteger()
+            this.kodeStatusKrav = status.toString()
+            this.kodeFagomraade = utledFagområdeKode(ytelsestype = ytelsestype)
+            this.fagsystemId = eksternFagsakId
+            this.vedtakIdOmgjort = BigInteger.ZERO
+            this.vedtakGjelderId = "12345678901"
+            this.typeGjelderId = TypeGjelderDto.PERSON
+            this.utbetalesTilId = "12345678901"
+            this.typeUtbetId = TypeGjelderDto.PERSON
+            this.enhetAnsvarlig = "8020"
+            this.enhetBosted = "8020"
+            this.enhetBehandl = "8020"
+            this.kontrollfelt = finalKontrollfelt
+            this.saksbehId = "K231B433"
+            this.referanse = eksternBehandlingId
+            this.tilbakekrevingsPeriode.addAll(tilbakekrevingsPerioder(antallPerioder = antallPerioder,
+                                                                       under4rettsgebyr = under4rettsgebyr,
+                                                                       muligforeldelse = muligforeldelse,
+                                                                       ytelsestype = ytelsestype,
+                                                                       periodeLengde = periodeLengde))
+        }
+    }
+
+    private fun utledFagområdeKode(ytelsestype: Ytelsestype): String {
+        return when (ytelsestype) {
+            Ytelsestype.BARNETRYGD -> "BA"
+            Ytelsestype.OVERGANGSSTØNAD -> "EFOG"
+            Ytelsestype.BARNETILSYN -> "EFBT"
+            Ytelsestype.SKOLEPENGER -> "EFSP"
+            Ytelsestype.KONTANTSTØTTE -> "KS"
+        }
+    }
+
+    private fun tilbakekrevingsPerioder(antallPerioder: Int,
+                                        under4rettsgebyr: Boolean,
+                                        muligforeldelse: Boolean,
+                                        ytelsestype: Ytelsestype,
+                                        periodeLengde: Int, ): List<DetaljertKravgrunnlagPeriodeDto> {
+        /*Lager alltid periodene 3 måneder lange med 1 måned mellom, så derfor gange med 4.
+        Første periode starter for 3 år siden når det skal være muligforeldelse eller for 4 måneder siden gange med antall perioder*/
```

Comment:
kommentar er for lang (mer enn 130 tegn)

Reply:
Endret kommentaren